### PR TITLE
fix: recover provider model discovery for Cline, MiMo Free, and xAI

### PR DESCRIPTION
**Pull request by Wibias**
Fixes model discovery so ClinePass and MiMo Free stop showing an HTTP 4xx discovery failure on hover, and adds recovery guidance to the Models page for providers whose discovery genuinely fails (for example xAI HTTP 400, prompting re-login for OAuth providers).

## What changed

- ClinePass and MiMo Free now use the maintained static catalog. Neither documents an OpenAI-style GET /models route, so the proxy no longer probes an unsupported endpoint and no longer misleads with a 4xx discovery failure.
- A registry-only capability flag disables the Discover models from provider toggle for these providers on the Models and Providers pages, with an explanation that a maintained static catalog is used.
- Models page recovery guidance: a failed discovery now shows the reason plus how to recover; OAuth providers (for example xAI) prompt to re-login, otherwise it points to provider settings and notes configured models stay available.
- MiMo Free legacy auth repair: a canonical row previously persisted as local is repaired to registry-owned key auth consistently across enrichment, request routing, and management DTOs, while custom destinations and explicit credentials are preserved.
- No invented capability for cline-pass/qwen3.8-max: the new model is no longer auto-classified as text-only / no-vision where the gateway does not document it.

## Verification

- Focused changed-path suites pass (ClinePass, MiMo Free, registry parity, router, management DTO, GUI models/discovery tests).
- typecheck, privacy scan, GUI tests, GUI lint, GUI i18n, GUI build, and docs site build all pass.
- The repository-wide bun run test on Windows is reported incomplete: an unrelated pre-existing storage-policy worker race timed out once and a Bun Windows runtime crash occurred in an aggregate run; that same storage test passed 3/3 in isolation and does not touch this diff.
- Security: Codex Security diff scan and an independent review both reported no findings; credential/destination boundaries and DTO redaction were specifically checked.

This PR is a draft and includes a UI screenshot of the Models discovery change.

## Summary by CodeRabbit

* **New Features**
  * Added maintained static model catalogs for providers without live discovery.
  * Added the ClinePass `qwen3.8-max` model.
  * Expanded discovery-failure guidance with fallback models and recovery actions, including reauthentication and settings access.
  * Added localized guidance across supported languages.

* **Bug Fixes**
  * Prevented unsupported model-list requests and network discovery for static providers.
  * Repaired legacy authentication and discovery settings automatically.
  * Preserved custom endpoints and configured model lists during provider updates.
  * Validated MiMo Free endpoints and provided clearer configuration errors.

### CODE REVIEW

**Comment by lidge-jun** [comment]
The static-catalog portion is soundly traced: `src/providers/derive.ts:397-405`, `src/router.ts:254-351`, and `src/codex/catalog/provider-fetch.ts:1095-1100` consistently force canonical ClinePass and MiMo Free rows to the maintained catalog without performing an unsupported `/models` request. The new route and management tests cover the legacy MiMo `local` repair, stale live-discovery state, custom-destination preservation, and DTO redaction. The completed Linux/macOS/Windows CI matrix is green, and no token or credential value is exposed by this diff.

[P1] `gui/src/pages/models-provider-hints.tsx:28-29` labels the failure CTA **"Open provider login"** for OAuth providers, but its handler is still only `navigateHash("providers")`. That opens the generic Providers page; it neither selects the failed provider nor starts its OAuth flow, so a user following the new xAI recovery instruction cannot actually re-login from this CTA. Either keep the existing "Open provider settings" label, or pass the provider identity and wire the action to the provider-specific login surface. Add a behavioral click test — `gui/tests/models-empty-provider.test.tsx` currently checks only rendered text, not where the CTA leads.

The PR also remains blocked by its failed `enforce-target` check: the repository requires a UI screenshot in the PR description, and committing `docs-site/public/pr-screenshots/provider-discovery-recovery.png` does not satisfy that. Embed the screenshot in the PR body, then rerun the gate and request the skipped CodeRabbit review.

DISPOSITION: NEEDS-CHANGE

File: gui/tests/models-empty-provider.test.tsx [hunk 1]
```diff
@@ -546,6 +550,16 @@ test("failed HTTP discovery renders an amber status badge and reason", () => {
...
+  expect(html).toContain("Configured models remain available");
+  expect(html).toContain("Check the endpoint or disable live model discovery");
+});
+
+test("OAuth discovery failure tells the user to re-login", () => {
+  const html = renderHint(true, { status: "failed", reason: "http", httpStatus: 400 }, "oauth");
+  expect(html).toContain("Re-login to restore live model discovery");
+  expect(html).toContain("Configured models remain available");
+  expect(html).toContain("Open provider settings");
+  expect(html).not.toContain("Open provider login");
```
[the remark this answers is not in the document]
Comment (Wibias) [reply gui/tests/models-empty-provider.test.tsx:563]: Would create an API mismatch.

**Review by lidge-jun — CHANGES_REQUESTED** [review]
[Repository bug audit · 2026-08-14]

The static-catalog/discovery recovery work is useful, but the reviewed diff contains a broader auth-mode behavior change that is not safely scoped to MiMo Free: `ProviderSettings` changes the fallback from `item.authMode ?? (item.keyOptional ? "local" : "key")` to `item.authMode ?? "key"` for every provider.

That can silently make other key-optional local/custom providers appear dirty or switch their UI default. Please keep the legacy repair at the canonical MiMo registry/server-enrichment boundary, or add an exhaustive provider-settings matrix proving no other key-optional provider changes behavior. Rebase afterward and rerun GUI provider-settings, registry parity, management DTO, model discovery, i18n, and build checks.

**Comment by lidge-jun** [comment]
Cherry-pick partially applied (test file resolved) but remaining commits conflict with provider discovery changes already in dev. Recommend a scoped re-implementation.
